src/Common/src/Interop/Unix/System.Native/Interop.FdSet.cs

```diff
@@ -2,34 +2,54 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
+using System.Diagnostics;
 using System.Runtime.InteropServices;
 
 internal static partial class Interop
 {
     internal static partial class Sys
     {
-        public unsafe struct FdSet
-        {
-            const int FDSET_MAX_FDS = 1024;
-            const int FDSET_NFD_BITS = FDSET_MAX_FDS / (8 * sizeof(uint));
+        [DllImport(Libraries.SystemNative)]
+        private static extern int FdSetSize();
+
+        private static readonly int FD_SETSIZE_BITS = FdSetSize();
+
+        private const int BitsPerByte = 8;
+        private static readonly int FD_SETSIZE_BYTES = FD_SETSIZE_BITS / BitsPerByte;
+
+        internal static readonly int FD_SETSIZE_UINTS = FD_SETSIZE_BYTES / sizeof(uint);
 
-            private fixed uint _bits[FDSET_MAX_FDS / FDSET_NFD_BITS];
-  
-            public void Set(int fd)
+        internal static unsafe void FD_SET(int fd, uint* fdset)
+        {
+            if (fd >= FD_SETSIZE_BITS)
```

Comment:
The original repro was a legitimate FD value because FD_SETSIZE had been increased to 64K, e.g. changing both the limit and the header and anything that depends on it, https://cs.uwaterloo.ca/~brecht/servers/openfiles.html.